crates/polars-plan/src/dsl/function_expr/nan.rs

```diff
@@ -4,12 +4,12 @@ pub(super) fn drop_nans(s: Series) -> PolarsResult<Series> {
     match s.dtype() {
         DataType::Float32 => {
             let ca = s.f32()?;
-            let mask = ca.is_not_nan();
+            let mask = ca.is_not_nan() | ca.is_null();
             ca.filter(&mask).map(|ca| ca.into_series())
         },
         DataType::Float64 => {
             let ca = s.f64()?;
-            let mask = ca.is_not_nan();
+            let mask = ca.is_not_nan() | ca.is_null();
```

Comment:
I think we can save a `negate + allocation` if we say `& ca.is_not_null()`. In that case we can use the validity mask directly and convert that to boolean array.

Reply:
Ah, nvm.. I now see the context. ^^